ParkingLotApiTest/ControllerTest/ParkingLotsControllerTests.cs

```diff
@@ -0,0 +1,249 @@
+        {
+            var client = GetClient();
+            var parkingLots = SeedParkingLots();
+            foreach (var lot in parkingLots)
+            {
+                var content = SerializeRequestBody(lot);
+                await client.PostAsync(RootUri, content);
+            }
+
+            var getResponse = await client.GetAsync($"{RootUri}?pageIndex=1");
+
+            getResponse.EnsureSuccessStatusCode();
+            var returnedParkingLots = await DeserializeResponseBodyAsync<List<ParkingLotDto>>(getResponse);
+            Assert.Equal(2, returnedParkingLots.Count);
+            Assert.Equal(parkingLots[0], returnedParkingLots[0]);
+            Assert.Equal(parkingLots[1], returnedParkingLots[1]);
+
+            var getResponse2 = await client.GetAsync($"{RootUri}?pageIndex=2");
+
+            getResponse2.EnsureSuccessStatusCode();
+            var returnedParkingLots2 = await DeserializeResponseBodyAsync<List<ParkingLotDto>>(getResponse2);
+            Assert.Empty(returnedParkingLots2);
+        }
+
+        [Fact]
+        public async Task Should_update_parking_lot_capacity_when_patch_update()
+        {
+            var client = GetClient();
+            var parkingLot = SeedParkingLot();
+            var parkingLotContent = SerializeRequestBody(parkingLot);
+            var createResponse = await client.PostAsync(RootUri, parkingLotContent);
+            createResponse.EnsureSuccessStatusCode();
+
+            var lotUpdate = new ParkingLotUpdateDto()
+            {
+                Capacity = 100,
+            };
+            var updateContent = SerializeRequestBody(lotUpdate);
+            var updateResponse = await client.PatchAsync(createResponse.Headers.Location, updateContent);
+
+            updateResponse.EnsureSuccessStatusCode();
+            var getResponse = await client.GetAsync(createResponse.Headers.Location);
+            getResponse.EnsureSuccessStatusCode();
+            var newParkingLot = await DeserializeResponseBodyAsync<ParkingLotDto>(getResponse);
+            Assert.Equal(lotUpdate.Capacity, newParkingLot.Capacity);
+        }
+
+        [Fact]
+        public async Task Should_not_update_parking_lot_capacity_when_patch_update_with_negative_capacity()
+        {
+            var client = GetClient();
+            var parkingLot = SeedParkingLot();
+            var parkingLotContent = SerializeRequestBody(parkingLot);
+            var createResponse = await client.PostAsync(RootUri, parkingLotContent);
+            createResponse.EnsureSuccessStatusCode();
+
+            var lotUpdate = new ParkingLotUpdateDto()
+            {
+                Capacity = -1,
+            };
+            var updateContent = SerializeRequestBody(lotUpdate);
+            var updateResponse = await client.PatchAsync(createResponse.Headers.Location, updateContent);
+
+            Assert.False(updateResponse.IsSuccessStatusCode);
```

Comment:
not sure if Assert.False(updateResponse.IsSuccessStatusCode) is enough to test error message, considering the requirement.